rotkehlchen/db/accounting_rules.py

```diff
@@ -204,6 +205,26 @@ def add_linked_setting(
                 f'Link of rule for {rule_property} to setting {setting_name} already exists',
             ) from e
 
+    def missing_accounting_rules(self, events: list[HistoryBaseEntry]) -> list[bool]:
+        """
+        For a list of events returns a list of the same length with boolean values where True
+        means that the event won't be affected by any accounting rule
+        """
+        query = 'SELECT COUNT(*) FROM accounting_rules WHERE (type=? AND subtype=? AND counterparty=?)'  # noqa: E501
+        bindings = [
+            (
+                event.event_type.serialize(), event.event_subtype.serialize(),
+                NO_ACCOUNTING_COUNTERPARTY if (counterparty := getattr(event, 'counterparty', None)) is None else counterparty,  # noqa: E501
+            ) for event in events
+        ]
+        missing_accounting_rule: list[bool] = []
+        with self.db.conn.read_ctx() as cursor:
+            for idx in range(len(events)):
+                row = cursor.execute(query, bindings[idx]).fetchone()
```

Comment:
wonder if there is any way to do this with a single execute. Can't use the `IN` check here.